### ISSUE TITLE
Failing on sync

### ISSUE THREAD

**Issue by SeungjinPark2** [wavesplatform/Waves#3669]
```
2022-04-06 07:31:28,883 INFO  [appender-24] c.w.s.BlockchainUpdaterImpl - New height: 3060352
2022-04-06 07:31:28,884 WARN  [rx-extension-loader-50] c.w.s.a.ExtensionAppender$ - [f95071a7 162.55.39.115:6868] Error appending extension (total=100) [BqbbiyP... -- 6MXCh4v...]: TransactionValidationError(cause = AccountBalanceError(Map(3PQiFvmHbqRskTjkU8u7oQ4cRi7uCogxn6b -> negative waves balance: 3PQiFvmHbqRskTjkU8u7oQ4cRi7uCogxn6b, old: 99000009, new: -5709694553)),
tx = {
  "type" : 7,
  "id" : "AhC49Bc5Tu9d6a5nhamDN3rJyWVdTFyADMVL1LKpAeAK",
  "fee" : 300000,
  "feeAssetId" : null,
  "timestamp" : 1649150642764,
  "version" : 2,
  "sender" : "3PEjHv3JGjcWNpYEEkif2w8NXV4kbhnoGgu",
  "senderPublicKey" : "9cpfKN9suPNvfeUNphzxXMjcnn974eme8ZhWUjaktzU5",
  "proofs" : [ "4tnppyQcarf6XunVeB5y2zdNh8zFybXKHBWfMQzgrzd86WXie8mf8GYYZesubNY15Lj7gFMboto1awx5xnsdWUgV" ],
  "order1" : {
    "version" : 3,
    "id" : "3ZWiGYjXXSUpZ8Ujb2wKimQxDSTJUwW2i5rS3TaXVSEX",
    "sender" : "3PPZWgFNRKHLvM51pwS934C8VZ7d2F4Z58g",
    "senderPublicKey" : "7JeravxHejNM5QqFiU1URLXYHueuDGzjdGucMBFJe5nz",
    "matcherPublicKey" : "9cpfKN9suPNvfeUNphzxXMjcnn974eme8ZhWUjaktzU5",
    "assetPair" : {
      "amountAsset" : null,
      "priceAsset" : "DG2xFkPdDwKUoBkzGAhQtLpSGzfXLiCYPEzeKH2Ad24p"
    },
    ...
```
Can not proceed synchronize after this log.
My waves version is 1.4.2
Environment (docker)
```
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=18.04
DISTRIB_CODENAME=bionic
DISTRIB_DESCRIPTION="Ubuntu 18.04.5 LTS"
NAME="Ubuntu"
VERSION="18.04.5 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.5 LTS"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic
```

**Comment by SeungjinPark2** [comment]
I'm trying 1.3.14 but also same.

**Comment by phearnot** [comment]
Hey, thanks for reaching out. Are you running the node built from `v1.4.2` tag, or is it some subsequent commit?
When checking if 1.3.14 works, did you use the same data directory as for 1.4.2, or did you start the import with a clean directory?

**Comment by SeungjinPark2** [comment]
I used 1.4.2 data directory when checking 1.3.14, did not clean my directory.
Why negative waves balance happens?
The transaction looks normal on explorer.

**Comment by phearnot** [comment]
There was an issue with 1.4 which incorrectly caused a transaction from the preceding block to fail, which in turn invalidated the mentioned exchange transaction. This is actually the reason why 1.3.14 didn't work from the same directory: the state had already been corrupted at height 3060350. One could roll back version 1.3.14 to 3060349 to successfully continue import.

 The issue is now fixed in #3597.
